File: backend_net/ChronosDtnNetApi/DTOs/LoginRequest.cs
// Por que: Importa o namespace System.ComponentModel.DataAnnotations para permitir validações de entrada via anotações.
using System.ComponentModel.DataAnnotations;

// Por que: Define o namespace DTOs para organizar objetos de transferência de dados da API.
namespace ChronosDtnNetApi.DTOs;

// Por que: Record imutável representando a requisição de login contendo as credenciais do operador.
public record LoginRequest(
    // Por que: O nome de usuário do operador deve ser fornecido obrigatoriamente.
    [Required(ErrorMessage = "Username é obrigatório")]
    string Username,

    // Por que: A senha de segurança deve ser fornecida obrigatoriamente.
    [Required(ErrorMessage = "Senha é obrigatória")]
    string Password
);

File: backend_java/src/main/java/br/com/fiap/chronosdtn/dto/LoginRequest.java
package br.com.fiap.chronosdtn.dto;

// Por que: Importa anotações de validação para impedir entradas vazias ou nulas no endpoint de autenticação.
import jakarta.validation.constraints.NotBlank;

// Por que: Usamos Record para garantir imutabilidade e thread-safety automática nos dados trafegados.
public record LoginRequest(
    // Por que: O nome de usuário do operador deve ser preenchido obrigatoriamente.
    @NotBlank(message = "Username é obrigatório")
    String username,

    // Por que: A senha do operador deve ser preenchida obrigatoriamente.
    @NotBlank(message = "Senha é obrigatória")
    String password
) {
}
